lisa/features/disks.py

```diff
@@ -64,6 +64,12 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
     def get_resource_disk_mount_point(self) -> str:
         raise NotImplementedError
 
+    # Check root partition's disk type to know the disk controller type
+    @property
+    def disk_controller_type(self) -> str:
```

Comment:
The disk_controller_type exists in disk settings already. How are they work together? If the asap is a capability of some vm size, asap should be a new disk_controller_type, instead of a new field on disk.

Reply:
@squirrelsc For clarification, disk_controller_type in the disk field is used to set disk_controller_type in the ARM template for SKUs with can be deployed with **either** SCSI or NVMe (ASAP) disk_controller_type.

For this test, Johnson suggested we check the OS disk type directly to determine the disk controller type. There are a few cases where this can be useful. For example, E112ibs_v5 only supports NVMe disk_controller_type. If disk field disk_dontroller_type is not set, we still need to detect that disk_controller_type is NVMe based on the OS disk type after deployment.

Reply:
I had a discussion with Srikanth. The value is used to decide the patterns to detect disks under `/dev`. So it just needs to add more patterns to include nvme devices by default.